src/engine/sparqlExpressions/ConditionalExpressions.cpp

```diff
@@ -42,30 +46,51 @@ class CoalesceExpression : public VariadicExpression {
     nextUnboundIndices.reserve(ctx->size());
 
     // Initially all result are unbound.
-    for (size_t i = 0; i < ctx->size(); ++i) {
-      // TODO<C++23> use `std::ranges::to<vector>(std::views::iota...)`.
-      unboundIndices.push_back(i);
-    }
+    ad_utility::chunkedForLoop<CHUNK_SIZE>(
+        0, ctx->size(),
+        [&unboundIndices](size_t i) {
+          // TODO<C++23> use `std::ranges::to<vector>(std::views::iota...)`.
+          unboundIndices.push_back(i);
+        },
```

Comment:
Thought about this for a while, I think the solution for ranges with known size would be to implement a for_each_chunked variant of the for_each function or something like that. But of course this would make the comment still redundant.